Assets/Scripts/Crafting/CraftingRecipeUI.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Constants;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Crafting
+{
+    public class CraftingRecipeUI : MonoBehaviour
+    {
+        [Tooltip("The name of this recipe")] [SerializeField]
+        private TextMeshProUGUI txtTitle;
+
+        [Tooltip("The vertical layout group where the resources will be ordered in")] [SerializeField]
+        private VerticalLayoutGroup recipeResourceList;
+
+        [Tooltip("The background image to show if the recipe can be crafted")] [SerializeField]
+        private Image imgBackground;
+
+        [Tooltip("The button for crafting the item")] [SerializeField]
+        private Button craftButton;
+
+        [Tooltip("The crafting recipe resource prefab")] [SerializeField]
+        private CraftingRecipeResource craftingRecipeResourcePrefab;
+
+        /// <summary>
+        /// The crafting UI reference that holds the item handler
+        /// </summary>
+        private CraftingUI _craftingUI;
+
+        /// <summary>
+        /// The recipe of this CraftingRecipeUI element
+        /// </summary>
+        private CraftingRecipe _recipe;
+
+        private void Start() => OnItemUpdate();
```

Comment:
this is called every time you open the crafting UI ? so it doesn't have to be updated? (i.e. when you gather a missing resource for the recipe, and the recipe should be craftable)

Reply:
ah didn't see ItemsUpdated l. 54 in InitRecipe :)

Reply:
This is only for triggering the update event on start. Makes sure it turns red from the beginning if you can't craft it and will be helpful later on when you load the game with items in your inventory